pkg/apimachinery/registered/registered.go

```diff
@@ -114,22 +114,24 @@ func IsEnabledVersion(v unversioned.GroupVersion) bool {
 	return found
 }
 
-// EnabledVersions returns all enabled versions.
-func EnabledVersions() (ret []unversioned.GroupVersion) {
-	for v := range enabledVersions {
-		ret = append(ret, v)
+// EnabledVersions returns all enabled versions.  Groups are randomly ordered, but versions within groups
+// are priority order from best to worst
+func EnabledVersions() []unversioned.GroupVersion {
+	ret := []unversioned.GroupVersion{}
+	for _, groupMeta := range groupMetaMap {
+		ret = append(ret, groupMeta.GroupVersions...)
```

Comment:
> Why are groups randomly ordered?  What value does it provide?

When choosing how to encode with a legacy encoder, the first matching group is used.  This is a natural method for getting all the GroupVersions and since the group will be traversed eventually, the group ordering doesn't matter.  The version ordering within that group is critical.